src/Wumpus.Net.Rest/Net/IDiscordRestApi.cs

```diff
@@ -43,9 +43,9 @@ internal interface IDiscordRestApi : IDisposable
         ///     If modifying a category, individual Channel Update events will fire for each child channel that also changes.
         /// </summary>
         [Patch("channels/{channelId}")]
-        Task<Channel> ModifyChannelAsync([Path] Snowflake channelId, [Body] ModifyChannelParams args);
+        Task<Channel> ModifyChannelAsync([Path] Snowflake channelId, [Body] ModifyChannelParams args, [Header(WumpusRestClient.ReasonHeader)] string reason = null);
```

Comment:
I think moving the header declaration to be a property akin to the authorization header would be more appropriate than passing it in as a method parameter. This would make it accessible immediately to any new endpoints that arise that can utilize it for audit logging.

The client would need to null out this property after every request though if changed to this, as not to reuse the same header value between requests, and...I'm not quite sure where that logic should go now that I look at what's possible with RestEase.

Reply:
Given that this property may change very frequently, I think this approach would also require considering thread saftey

Reply:
Forgot to consider that. Do you know if the endpoints you've added the header param for are the only ones that actually do anything if you specify the audit log header?

Reply:
[I went off of the list of events in the Audit Log documentation and tried to match each one.](https://discordapp.com/developers/docs/resources/audit-log#audit-log-entry-object-audit-log-events) I have not tested each one individually.